processing/src/main/java/io/druid/query/timeseries/TimeseriesQuery.java

```diff
@@ -64,11 +63,10 @@ public TimeseriesQuery(
       @JsonProperty("context") Map<String, Object> context
   )
   {
-    super(dataSource, querySegmentSpec, descending, context);
+    super(dataSource, querySegmentSpec, descending, context, granularity);
```

Comment:
How do timeseries queries behave with `null` granularity?

Reply:
I just tested and it throws a null point exception, moving the precondition check from group by into `TimeBucketedQuery` should fix this.

Reply:
It appears `TimeseriesQueryBuilder` defaults to `Granularities.ALL` which is why all of the tests which do not specify a granularity do not explode.